pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -104,22 +105,20 @@ func (w *Watcher) Start() error {
 
 	// Traverse plugin dir and add filesystem watchers before starting the plugin processing goroutine.
 	if err := w.traversePluginDir(w.path); err != nil {
-		w.Stop()
+		w.fsWatcher.Close()
 		return fmt.Errorf("failed to traverse plugin socket path %q, err: %v", w.path, err)
 	}
 
 	// Traverse deprecated plugin dir, if specified.
 	if len(w.deprecatedPath) != 0 {
 		if err := w.traversePluginDir(w.deprecatedPath); err != nil {
-			w.Stop()
+			w.fsWatcher.Close()
```

Comment:
Why do not use w.Stop? w.stopped chan seems not be closed?

Reply:
if we call Stop here it will block until the time out. chans don't need to be closed unless something is waiting in them. this one will go out of scope and be garbage collected. https://groups.google.com/forum/m/#!msg/golang-nuts/pZwdYRGxCIk/qpbHxRRPJdUJ

Reply:
> if we call Stop here it will block until the time out. chans don't need to be closed unless something is waiting on them. stopped will go out of scope and be garbage collected. https://groups.google.com/forum/m/#!msg/golang-nuts/pZwdYRGxCIk/qpbHxRRPJdUJ
> 
> > it is only necessary to close a channel if the receiver is
> > looking for a close.  Closing the channel is a control signal on the
> > channel indicating that no more data follows.

Thanks for your clarification!